2_attributePoints.R

```diff
@@ -119,49 +120,77 @@ fullL <- gridlist[names(gridlist) %in% tolower(gridlistSub$gridName)]
+raster_names<-names(fullL)
+
+#Notes on trouble shooting stars from PJM
+#temp_stars<-read_stars(raster_paths[1:38]) #works
+#temp_stars<-read_stars(raster_paths[1:40]) #beersx1000 & crvslpx100 are exmples that won't read in with others
+#temp_stars<-read_stars(raster_paths[39:40])
+#stars_1<-read_stars(raster_paths[1]) #clim_bio1
+#stars_39<-read_stars(raster_paths[39]) #beersx1000
+#combine_1_39<-c(stars_1, stars_39, along=1)
+#combine_1_39  #results printed to screen sugest possibly different CRS for some rasters?
 # extract raster data to points ----
 
+rm(fullL, gridlistSub, modType, branches, activeBranch)
+
 # Extract values to a data frame - multicore approach using snowfall
 # First, convert raster stack to list of single raster layers
+#PJM: slist not utilized in current script, nor is envStack
 s.list <- unstack(envStack)
 names(s.list) <- names(envStack)
 
+# PJM: commented this bit about break out to try looping through rasters with stars package
 # with contributions from PJM
 # if more than 10k points, split into groups approx 5k each
-if(nrow(shpf) > 10000){
-  shpRows <- 1:nrow(shpf)
-  numBrks <- ceiling(nrow(shpf)/5000)
-  brks <- cut(shpRows, breaks = numBrks)
-  brkgrps<- unique(brks)  
-} else {
-  brks <- rep(1, nrow(shpf))
-  brkgrps <- 1
-}
+# if(nrow(shpf) > 10000){
+#   shpRows <- 1:nrow(shpf)
+#   numBrks <- ceiling(nrow(shpf)/5000)
+#   brks <- cut(shpRows, breaks = numBrks)
+#   brkgrps<- unique(brks)  
+# } else {
+#   brks <- rep(1, nrow(shpf))
+#   brkgrps <- 1
+# }
 
 #debug/checking
-print(paste0("num gps for attribute: ", length(brkgrps)))
+#print(paste0("num gps for attribute: ", length(brkgrps)))
 
 e.df_list<-list()
 # start cluster
-sfInit(parallel=TRUE, cpus=parallel:::detectCores()-1)
-sfLibrary(raster)
-sfLibrary(sf)
+#sfInit(parallel=TRUE, cpus=parallel:::detectCores()-1)
+#sfLibrary(raster)
+#sfLibrary(sf)
 # extract by subsets to keep RAM from maxing out
-for(i in 1:length(brkgrps)){
-  shpf_sub<-shpf[brks==brkgrps[i],]
+for(i in 1:length(raster_paths)){
+  stars_rast<-read_stars(raster_paths[i])  #reads in rasters individually as stars object or stars_proxy depending on size.  
   # Run parallelized 'extract' function
-  e.df_list[[i]] <- sfSapply(s.list, extract, y=shpf_sub, method = "simple")
+  #e.df_list[[i]] <- sfSapply(s.list, extract, y=shpf_sub, method = "simple")
+  #e.df_list[[i]] <- sfSapply(temp_stars, st_extract, y=shpf) #tried this on a group of stars_objects
+  
+  e.df_extract <- st_extract(stars_rast,shpf)  #uses st_extract function to extract raster data to points
+  e.df_extract<-as.data.frame(e.df_extract)   #this bit was to get rid of the geometry  stuck on the end 
+  e.df_extract<-subset(e.df_extract, select=-geometry)  #this bit was to get rid of the geometry object stuch on the end 
+  e.df_list[[i]]<-as.data.frame(e.df_extract)   #
 }
 # stop cluster
-sfStop()
+#sfStop()
 
-e.df<-do.call(rbind, e.df_list)
 
+
+e.df<-do.call(rbind, e.df_list)
```

Comment:
I need to use `cbind` here, in my test

Reply:
can you confirm?

Reply:
yes- Thought I'd fixed that but must not have pushed that particular change
should be cbind